ui/admin/app/components/form/target/details/index.hbs

```diff
@@ -75,86 +81,141 @@
     {{#if @model.errors.session_connection_limit}}
-      <field.errors as |errors|>
+      <F.Error as |E|>
         {{#each @model.errors.session_connection_limit as |error|}}
-          <errors.message>{{error.message}}</errors.message>
-        {{/each}}
-      </field.errors>
-    {{/if}}
-  </form.input>
-
-  <form.input
-    @name='worker_filter'
-    @type='text'
-    @value={{@model.worker_filter}}
-    @label={{t 'form.worker_filter.label'}}
-    @error={{@model.errors.worker_filter}}
-    @helperText={{t 'form.worker_filter.help'}}
-    @linkText={{t 'actions.learn-more'}}
-    @link={{doc-url 'target.worker-filters'}}
-    as |field|
-  >
-    {{#if @model.errors.worker_filter}}
-      <field.errors as |errors|>
-        {{#each @model.errors.worker_filter as |error|}}
-          <errors.message>{{error.message}}</errors.message>
+          <E.Message>{{error.message}}</E.Message>
         {{/each}}
-      </field.errors>
+      </F.Error>
     {{/if}}
-  </form.input>
+  </Hds::Form::TextInput::Field>
 
-  <form.input
-    @name='default_port'
-    @type='number'
+  <Hds::Form::TextInput::Field
+    @isRequired={{false}}
+    @isOptional={{true}}
     @value={{@model.default_port}}
-    @label={{t 'form.default_port.label'}}
-    @error={{@model.errors.default_port}}
-    @helperText={{t 'form.default_port.help'}}
-    placeholder={{@defaultPort}}
-    min='0'
-    step='1'
-    as |field|
+    @isInvalid={{@model.errors.default_port}}
+    @type='text'
+    name='default_port'
+    disabled={{form.disabled}}
+    {{on 'input' (set-from-event @model 'default_port')}}
+    as |F|
   >
+    <F.Label>{{t 'form.default_port.label'}}</F.Label>
+    <F.HelperText>{{t 'form.default_port.help'}}</F.HelperText>
     {{#if @model.errors.default_port}}
-      <field.errors as |errors|>
+      <F.Error as |E|>
         {{#each @model.errors.default_port as |error|}}
-          <errors.message>{{error.message}}</errors.message>
+          <E.Message>{{error.message}}</E.Message>
         {{/each}}
-      </field.errors>
+      </F.Error>
     {{/if}}
-  </form.input>
+  </Hds::Form::TextInput::Field>
+
+  {{#if (feature-flag 'target-worker-filters-v2')}}
+    <div class='target-workers'>
+      <Hds::Form::Legend class='legend'>{{t
+          'resources.target.workers.title'
+        }}</Hds::Form::Legend>
+      <Hds::Form::HelperText class='description'>{{t
+          'resources.target.workers.description'
+        }}</Hds::Form::HelperText>
+      {{#if this.showDeprecationMessage}}
+        <Hds::Alert
+          class='deprecation-message'
+          @type='inline'
+          @icon='info'
+          as |A|
+        >
+          <A.Title>{{t
+              'resources.target.workers.deprecation_message.title'
+            }}</A.Title>
+          <A.Description>{{t
+              'resources.target.workers.deprecation_message.description'
+            }}</A.Description>
+        </Hds::Alert>
+      {{/if}}
+      <Hds::Form::TextInput::Field
```

Comment:
Should this field be shown at all if we're not showing the deprecation message (i.e. the field shouldn't show if there is no worker filter)? Wouldn't we then normally be showing the ingress/egress filters so this worker filter should be part of the `showDeprecationMessage` if check?

Reply:
Yes absolutely. I left it for now as a kind of place holder until I add the `egress` filter in the next story. I felt leaving a disabled input field looked better than just a "Workers" section with a description with no field at all. I am happy to move it within the check now but I will be updating it soon.